server/mainPr/src/main/java/com/team23/mainPr/Domain/RentPost/Service/RentPostService.java

```diff
@@ -134,4 +134,13 @@ public PagedRentPostResponseDtos getRentPosts(Pageable pageable, Boolean rentSta
         return response;
     }
 
+    public List<RentPostResponseDto> searchAll(String phrase) {
+        List<RentPostResponseDto> result = new ArrayList<>();
+        rentPostRepository.search("" + phrase.replace(" ", "|").trim() + "")
+                .stream().forEach(
+                        rentPostId ->{
+                            result.add(rentPostMapper.RentPostToRentPostResponseDto(rentPostRepository.getReferenceById((Integer) rentPostId)));
+                        });
+        return result;
+    }
```

Comment:
```suggestion
    public List<RentPostResponseDto> searchAll(String phrase) {
        return rentPostRepository.search(phrase.replace(" ", "|").trim()).stream()
                .map(e -> rentPostMapper.RentPostToRentPostResponseDto(rentPostRepository.getReferenceById((Integer) e)))
                .collect(Collectors.toList());
    }
```

stream 을 활용해서 개선한 코드도 별로입니다.  좀 더 고민해주세요.

Reply:
옵셔널을 없애고 바로 List<Integer>로 응답을 받아, collect 함수로 바로 리턴 하도록 리팩토링 하였습니다.

Reply:
프론트에서 검색어를 메소드에 필요한 형태( phrase.replace(" ", "|").trim() ) 로 보내도록 요청하거나 컨트롤러에서 변경 후에 서비스에 요청하도록 변경하는게 맞는 것 같습니다.

Reply:
RentPostRepository 의 쿼리 수행 메서드의 응답 결과를 ```List<Object>``` 가 아닌 ```List<RentPost>```로 바로 엔티티로 맵핑이 되도록 하였고, collect 함수를 사용하여 한번에 응답할 수 있도록 리팩토링 하였습니다.